src/sql/workbench/parts/notebook/notebook.component.ts

```diff
@@ -225,13 +235,38 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 			await this.setNotebookManager();
 			await this.loadModel();
 			this._modelReadyDeferred.resolve(this._model);
-		} catch (error) {
-			this.setViewInErrorState(localize('displayFailed', 'Could not display contents: {0}', notebookUtils.getErrorMessage(error)));
-			this.setLoading(false);
-			this._modelReadyDeferred.reject(error);
-		} finally {
-			// Always add the editor for now to close loop, even if loading contents failed
 			this.notebookService.addNotebookEditor(this);
+		} catch (error) {
+			// Offer to create a file from the error if we have a file not found and the name is valid
+			if ((<FileOperationError>error).fileOperationResult === FileOperationResult.FILE_NOT_FOUND && isValidBasename(basename(this.notebookParams.notebookUri))) {
```

Comment:
So if it's not a FileOperationError it'll just return undefined for `fileOperationResult` and fail comparison without getting null ref?